core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -921,9 +912,9 @@ private object RTS {
     final def shouldDie: Option[Throwable] =
       if (!killed || noInterrupt > 0) None else status.get.error
 
-    final def enterUninterruptible: IO[E, Unit] = IO.sync { noInterrupt += 1 }
+    final def enterUninterruptible[E]: IO[E, Unit] = IO.sync { noInterrupt += 1 }
```

Comment:
For performance, turn these into `final val enterUninterruptible: IO[E, Void] ...`

Same for `exitUninterruptibble`. Then you can call `widenError` on these later.

Then there will be only one allocation, rather than one each time it is called. The tradeoff is higher initial allocation.